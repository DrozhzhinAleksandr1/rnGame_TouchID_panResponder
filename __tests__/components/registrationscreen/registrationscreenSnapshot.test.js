import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { RegistrationScreen } from '../../../app/components/registrationscreen/registrationscreen';
import { localization } from '../../../app/common/localization';
import stackNav from '../../../app/redux/reducers/stacknavreducer';
stackNav.setParams = () => {
    return "test";
}
mock = () => { };
it('create snapshot RegistrationScreen', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<RegistrationScreen CurrentLanguage={localization.ru} navigation={stackNav}
        setRegistrationLastName={mock}
        setRegistrationFirstName={mock}
        setRegistrationEmail={mock}
        setRegistrationCountry={mock}
        setRegistrationPhone={mock}
        validObjCheckboxgistration={false} />);
    expect(toolbarSnapshot).toMatchSnapshot();
});
