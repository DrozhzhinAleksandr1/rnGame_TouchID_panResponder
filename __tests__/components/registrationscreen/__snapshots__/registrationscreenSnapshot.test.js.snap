// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot RegistrationScreen 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#172533",
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <withNavigation(Connect(ThanksForJoiningPopUp)) />
  <ScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "flex": 1,
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "marginBottom": 25,
          "marginTop": 25,
          "width": 296,
        }
      }
    >
      <Input
        accessibilityLabel="InputFirstNameRegistrationID"
        action={[Function]}
        errorText="Ошибка ввода имени"
        filter={/\\(\\?=\\(\\^\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#'!\\?\\$%\\^&\\*_=\\+-\\]\\)\\)\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:@"#!\\?\\$%\\^&\\*_=\\+\\]\\{0,199\\}\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#!\\?\\$%'\\^&\\*_=\\+-\\]\\$\\)/}
        placeholder="Имя"
        testErrorID="FirstNameErrorRegostrationID"
        testID="InputFirstNameRegistrationID"
      />
      <Input
        accessibilityLabel="InputLastNameRegistrationID"
        action={[Function]}
        errorText="Ошибка ввода фамилии"
        filter={/\\(\\?=\\(\\^\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#'!\\?\\$%\\^&\\*_=\\+-\\]\\)\\)\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:@"#!\\?\\$%\\^&\\*_=\\+\\]\\{0,199\\}\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"#!\\?\\$%'\\^&\\*_=\\+-\\]\\$\\)/}
        placeholder="Фамилия"
        testErrorID="LastNameErrorRegostrationID"
        testID="InputLastNameRegistrationID"
      />
      <Input
        accessibilityLabel="InputEmailRegistrationID"
        action={[Function]}
        errorText="Ошибка ввода e-mail"
        filter={/\\^\\(\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\(\\\\\\.\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\)\\*\\)\\|\\("\\.\\+"\\)\\)@\\(\\(\\\\\\[\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\]\\)\\|\\(\\(\\[a-zA-Z\\\\-0-9\\]\\+\\\\\\.\\)\\+\\[a-zA-Z\\]\\{2,\\}\\)\\)\\$/}
        placeholder="Email"
        testErrorID="EmailErrorRegostrationID"
        testID="InputEmailRegistrationID"
      />
      <withNavigation(CountryList)
        textTitle="Страна"
      />
      <Connect(PhoneInput) />
      <withNavigation(AgreementBtn)
        textTitle="Заполняя форму регистрации,"
        textWithClick="я принимаю условия обработки персональных данных."
      />
      <FormBtnSubmit
        accessibilityLabel="BtnRegistrationInRegistrationID"
        func={[Function]}
        isEnable={false}
        testID="BtnRegistrationInRegistrationID"
        text="Зарегистрироваться"
      />
      <withNavigation(AuthorizationBtn)
        textLogin="Войти"
        textTitle="Уже есть аккаунт ?"
      />
    </Component>
  </ScrollView>
  <Connect(ErrorValidation)
    testIDBtn="RegistrationBtnId"
    testIDText="RegistrationTextId"
  />
</Component>
`;
